src/Nethermind/Nethermind.Merge.Plugin/GC/GCKeeper.cs

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Runtime;
+using System.Threading.Tasks;
+using FastEnumUtility;
+using Nethermind.Core.Extensions;
+using Nethermind.Logging;
+
+namespace Nethermind.Merge.Plugin.GC;
+
+public class GCKeeper
+{
+    private readonly IGCStrategy _igcStrategy;
+    private readonly ILogger _logger;
+    private static readonly long _defaultSize = 512.MB();
+    private Task _gcScheduleTask = Task.CompletedTask;
+
+    public GCKeeper(IGCStrategy igcStrategy, ILogManager logManager)
+    {
+        _igcStrategy = igcStrategy;
+        _logger = logManager.GetClassLogger<GCKeeper>();
+    }
+
+    public IDisposable TryStartNoGCRegion(long? size = null)
+    {
+        size ??= _defaultSize;
```

Comment:
Can add a flag to completely disable this behaviour if needed?

Reply:
yes was thinking about it